pkg/services/thumbs/crawler.go

```diff
@@ -83,61 +77,31 @@ func (r *simpleCrawler) broadcastStatus() {
 	}
 }
 
-func (r *simpleCrawler) GetPreview(req *previewRequest) *previewResponse {
-	p := getFilePath(r.screenshotsFolder, req)
-	if _, err := os.Stat(p); errors.Is(err, os.ErrNotExist) {
-		return r.queueRender(p, req)
-	}
-
-	return &previewResponse{
-		Path: p,
-		Code: 200,
-	}
-}
-
-func (r *simpleCrawler) queueRender(p string, req *previewRequest) *previewResponse {
-	go func() {
-		fmt.Printf("todo? queue")
-	}()
-
-	return &previewResponse{
-		Code: 202,
-		Path: p,
-	}
-}
-
-func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme rendering.Theme) (crawlStatus, error) {
-	if r.status.State == "running" {
+func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme models.Theme, thumbnailKind models.ThumbnailKind) (crawlStatus, error) {
+	if r.status.State == running {
```

Comment:
I think this check should be locked. what about using `r.Status()` instead?

Reply:
yeah, I have already changed it in a different PR that targets this branch: https://github.com/grafana/grafana/pull/44891